modules/custom/az_flexible_page/src/Form/AzFlexiblePageSettingsForm.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\az_flexible_page\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteBuilderInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form for custom AZ Flexible Page module settings.
+ */
+class AzFlexiblePageSettingsForm extends ConfigFormBase {
+
+  /**
+   * The route builder.
+   *
+   * @var \Drupal\Core\Routing\RouteBuilderInterface
+   */
+  protected $routeBuilder;
+
+  /**
+   * Constructs a AzFlexiblePageSettingsForm object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   * @param \Drupal\Core\Routing\RouteBuilderInterface $route_builder
+   *   The route builder.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, RouteBuilderInterface $route_builder) {
+    parent::__construct($config_factory);
+
+    $this->routeBuilder = $route_builder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('router.builder')
+    );
+  }
```

Comment:
Since I don't think the routeBuilder call in the submit handler is actually needed for this form, I don' t think any of this is needed, either.
```suggestion
```